### ISSUE TITLE
Error: Metadata version mismatch for module .../node_modules/@ng-select/ng-select/ng-select.d.ts

### ISSUE THREAD

**Issue by jericdeleon** [ng-select/ng-select#329]
### Expected behaviour
`ng serve` starts an Angular application (using ng-select) with no problems

### Actual behaviour
Upon `ng serve`, error occurs:
```
> sample@0.0.1 start /home/jericdeleon/Documents/sample                                                                                                                                                                    
> ng serve -H 0.0.0.0                                                                                                                                                                                                                                         
                                                                                                                                                                                                                                                              
** NG Live Development Server is listening on 0.0.0.0:4200, open your browser on http://localhost:4200/ **                                                                                                                                                    
Date: 2018-03-08T19:37:41.447Z                                                                                                                                                                                                                                
Hash: 6a0a086e69ec17a39f70                                                                                                                                                                                                                                    
Time: 15764ms                                                                                                                                                                                                                                                 
chunk {inline} inline.bundle.js (inline) 3.85 kB [entry] [rendered]                                                                                                                                                                                           
chunk {main} main.bundle.js (main) 788 kB [initial] [rendered]                                                                                                                                                                                                
chunk {polyfills} polyfills.bundle.js (polyfills) 529 kB [initial] [rendered]                                                                                                                                                                                 
chunk {styles} styles.bundle.js (styles) 587 kB [initial] [rendered]                                                                                                                                                                                          
chunk {vendor} vendor.bundle.js (vendor) 14.2 MB [initial] [rendered]                                                                                                                                                                                         
                                                                                                                                                                                                                                                              
ERROR in Error: Metadata version mismatch for module /home/jericdeleon/Documents/sample/node_modules/@ng-select/ng-select/ng-select.d.ts, found version 4, expected 3, resolving symbol SharedModule in /home/jericdeleon/D
ocuments/sample/src/app/shared/index.ts, resolving symbol SharedModule in /home/jericdeleon/Documents/sample/src/app/shared/index.ts                                                    
    at syntaxError (/home/jericdeleon/Documents/sample/node_modules/@angular/compiler/bundles/compiler.umd.js:1729:34)                                                                                                     
    at simplifyInContext (/home/jericdeleon/sample/node_modules/@angular/compiler/bundles/compiler.umd.js:24979:23)                                                                                              
    at StaticReflector.simplify (/home/jericdeleon/sample/node_modules/@angular/compiler/bundles/compiler.umd.js:24991:13)                                                                                       
    at StaticReflector.annotations (/home/jericdeleon/sample/node_modules/@angular/compiler/bundles/compiler.umd.js:24418:41)                                                                                    
    at _getNgModuleMetadata (/home/jericdeleon/sample/node_modules/@angular/compiler-cli/src/ngtools_impl.js:138:31)                                                                                             
    at _extractLazyRoutesFromStaticModule (/home/jericdeleon/sample/node_modules/@angular/compiler-cli/src/ngtools_impl.js:109:26)                                                                               
    at /home/jericdeleon/sample/node_modules/@angular/compiler-cli/src/ngtools_impl.js:129:27                                                                                                                    
    at Array.reduce (<anonymous>)                              
    at _extractLazyRoutesFromStaticModule (/home/jericdeleon/sample/node_modules/@angular/compiler-cli/src/ngtools_impl.js:128:10)                                                                               
    at /home/jericdeleon/sample/node_modules/@angular/compiler-cli/src/ngtools_impl.js:129:27                                                                                                                    
    at Array.reduce (<anonymous>)                              
    at _extractLazyRoutesFromStaticModule (/home/jericdeleon/sample/node_modules/@angular/compiler-cli/src/ngtools_impl.js:128:10)                                                                               
    at Object.listLazyRoutesOfModule (/home/jericdeleon/sample/node_modules/@angular/compiler-cli/src/ngtools_impl.js:53:22)                                                                                     
    at Function.NgTools_InternalApi_NG_2.listLazyRoutes (/home/jericdeleon/sample/node_modules/@angular/compiler-cli/src/ngtools_api.js:91:39)                                                                   
    at AotPlugin._getLazyRoutesFromNgtools (/home/jericdeleon/sample/node_modules/@ngtools/webpack/src/plugin.js:240:66)                                                                                         
    at _donePromise.Promise.resolve.then.then.then.then.then (/home/jericdeleon/sample/node_modules/@ngtools/webpack/src/plugin.js:493:24)
```

### More Info
Metadata version error doesn't appear on `0.23.1`, but errors out on every version on or after `0.23.2`.
A sample repo with the issue can be found here: https://github.com/jericdeleon/angularspree/tree/ng-select-metadata-error
(the `HEAD` commit shows the error; using ng-select `0.23.1` or below makes it work for now. `HEAD~1` is the previous working version, prior to adding ng-select.). Feel free to clone and run `npm run start`.

**Comment by maxkorz** [comment]
In version 0.23.2 [ng-packagr was updated to version "^2.1.0"](https://github.com/ng-select/ng-select/pull/304/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2L66).

[Version 2 of ng-packagr is targeting Angular 5 (and beyond).](https://github.com/dherges/ng-packagr/blob/master/CHANGELOG.md#200-2018-02-05). You need to update your project to Angular 5.x

**Comment by jericdeleon** [comment]
Glad to know this, thanks for the timely help!

**Comment by prabhu2289** [comment]
Is there any other way to run in angular 4?

**Comment by varnastadeus** [comment]
@prabhu2289 you could use older version which is highly unrecommended since you will miss all bug fixes and new features.

**Comment by Dima11235813** [comment]
This worked for the longest time and then recently I had to reclone my repo and when I ran npm install this issue came up. 

I don't have time to upgrade to angular 5 right now, is there any other way around this as a temporary fix?

I found the old node_modules folder for this package and moved it to my new repo build and everything works.

A bit confused why an old version would be intentionally updated to not work without upgrading, what's going on here?
